qtglean/include/glean/uuid.h

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef UUID_H
+#define UUID_H
+#include <QObject>
+
+#include "errortype.h"
+
+class UuidMetric final {
+  Q_GADGET
+
+ public:
+  // QML custom types require these three declarations.
+  // See: https://doc.qt.io/qt-6/custom-types.html#creating-a-custom-type
+  UuidMetric() = default;
```

Comment:
Question for @brizental : can js/qml actually create these objects directly?

Reply:
Nope, this is not exposed to QML that way. This is required for the expansion of the Q_GADGET macro.